### ISSUE TITLE
samba-build: Failure in building packages off older hash.

### ISSUE THREAD

**Issue by spuiuk** [gluster/samba-integration#167]
I have been attempting to create centos8 packages for an older git hash to test for O_PATH issues.

# make rpms.centos8 refspec=h:7a077f152aa3e1e09b4228ae24780fa613540a68
..
TASK [build.rpms.centos8 : build centos 8 rpms] **************************************************************************************************************
fatal: [localhost]: FAILED! => {"changed": true, "cmd": ["/usr/bin/mock", "--root", "/tmp/samba-build/mock/epel-8-x86_64.cfg", "--enablerepo=Devel", "--resultdir", "/tmp/samba-build/rpms/master/8/x86_64", "--rebuild", "/tmp/samba-build/srpms/samba-20201215.153218.7a077f152aa-1.fc33.1.src.rpm"], "delta": "0:14:57.301675", "end": "2021-05-12 23:28:30.293528", "msg": "non-zero return code", "rc": 1, "start": "2021-05-12 23:13:32.991853", "stderr": "INFO: mock.py version 2.10 starting (python version = 3.9.2, NVR = mock-2.10-1.fc33)...\nStart(bootstrap): init plugins\nINFO: selinux enabled\nFinish(bootstrap): init plugins\nStart: init plugins\nINFO: selinux enabled\nFinish: init plugins\nINFO: Signal handler active\nStart: run\nINFO: Start(/tmp/samba-build/srpms/samba-20201215.153218.7a077f152aa-1.fc33.1.src.rpm)  .......

Looking through the build log, I see 

Processing files: samba-client-20201215.153218.7a077f152aa-1.el8.1.x86_64
error: File not found: /builddir/build/BUILDROOT/samba-20201215.153218.7a077f152aa-1.el8.1.x86_64/usr/bin/mdsearch
error: File not found: /builddir/build/BUILDROOT/samba-20201215.153218.7a077f152aa-1.el8.1.x86_64/usr/libexec/samba/samba-bgqd
error: File not found: /builddir/build/BUILDROOT/samba-20201215.153218.7a077f152aa-1.el8.1.x86_64/usr/share/man/man1/mdsearch.1*
error: File not found: /builddir/build/BUILDROOT/samba-20201215.153218.7a077f152aa-1.el8.1.x86_64/usr/share/man/man8/samba-bgqd.8*

I am assuming that this is caused because the spec file lists files which are not available in this version of upstream samba.

**Comment by spuiuk** [comment]
Tested using PR #165 still fails with the following list

    File not found: /builddir/build/BUILDROOT/samba-20201215.153218.7a077f152aa-1.el8.1.x86_64/usr/bin/mdsearch
    File not found: /builddir/build/BUILDROOT/samba-20201215.153218.7a077f152aa-1.el8.1.x86_64/usr/libexec/samba/samba-bgqd
    File not found: /builddir/build/BUILDROOT/samba-20201215.153218.7a077f152aa-1.el8.1.x86_64/usr/share/man/man1/mdsearch.1*
    File not found: /builddir/build/BUILDROOT/samba-20201215.153218.7a077f152aa-1.el8.1.x86_64/usr/share/man/man8/samba-bgqd.8*

**Comment by anoopcs9** [comment]
Yeah.. lack of spec file with upstream sources is the problem here. We don't have a history of upstream changes w.r.t spec file as they don't reside at same place.
